provider_shopper/lib/models/catalog.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+
+/// A proxy of the catalog of items the user can buy.
+///
+/// In a real app, this might be backed by a backend and cached on device.
+/// In this sample app, the catalog is procedurally generated.
+///
+/// For simplicity, the catalog is expected to be immutable (no products are
+/// expected to be added, removed or changed during the execution of the app).
+class CatalogModel {
+  static const _itemNames = [
+    'Code Smell',
+    'Control Flow',
+    'Interpreter',
+    'Recursion',
+    'Sprint',
+    'Heisenbug',
+    'Spaghetti',
+    'Hydra Code',
+    'Off-By-One',
+    'Scope',
+    'Callback',
+    'Closure',
+    'Automata',
+    'Bit Shift',
+    'Currying',
+  ];
+
+  /// Get item by [id].
+  Item getById(int id) => Item(id, _itemNames[id % _itemNames.length]);
```

Comment:
The modulo operation in here was confusing at first glance. Now that I've seen the rest of the code, I understand that you're doing it so that you have an infinitely scrolling list. That's a cool piece of UI to show, but I'm concerned that people won't be able to understand this file until they go further into the app.

Consider:

* Removing the infinite list and just showing the items in the catalog once.
* Commenting here and in the colors file to explain that you're faking an infinite list of products by repeating names and colors, and that's why the modulo operation is there (instead of a thrown IndexOutOfRange exception or something).